src/db.rs

```diff
@@ -283,6 +284,12 @@ pub struct Group {
 
     /// The list of entries in this group
     pub entries: IndexMap<String, Entry>,
+
+    /// The list of time fields for this group
+    pub times: HashMap<String, chrono::NaiveDateTime>,
```

Comment:
Are there other times variables other than ExpiryTime possible? Update: Yes, for creation, last modified, etc.

From a developer usability perspective, I would prefer `expires` to be an `Option<chrono::DateTime>` but I understand that it is mostly the KDBX format that is strange here. I leave it up to you what solution you like better.

Reply:
I coded expires as an Option<> in one of my earlier iterations; however, as you already discovered, there are several timestamps stored in the XML for each Entry and Group (as well as for Database and maybe other XML nodes) which might be useful to others.  Some examples, not an exhaustive list, these may not be the exact XML node names but are close:
CreatedTime
LastAccessedTime
LastModifiedTime

I am only currently populating ExpiryTime for a few reasons:
To minimize size of Database created by keepass::Database::open
To minimize XML parsing time for keepass::Database::open
To minimize my coding and testing effort before you had reviewed this approach

My thought was storing these as a HashMap<> (similar to Entry.fields) sets the stage for quick and easy addition of other XML timestamp fields as needed/requested or if you decide to make the keepass::Database a more exhaustive representation of the XML in the future.

Adding the getter convenience methods (get_expiry_time, etc.) seems like it would provide similar developer usability as using individual attributes.

If you would rather have all timestamps stored unique Option<> attributes rather than in one big HashMap, I can make that change.  Please let me know.

Also, do you want me to add any other timestamps to Entry and/or Group at this time?

Reply:
OK, that absolutely makes sense. Let's keep it as it is right now. You don't need to add the other timestamps right now, since it is not something that you need for your own use case. I can see that @Narigo might want to know about Entry modification timestamps for keepass-diff for a future automatic conflict resolution, but let's keep that for a future PR.